### ISSUE TITLE
Socket.io : Could not load session for socket #.......

### ISSUE THREAD

**Issue by billyshena** [balderdashy/sails#2263]
Hello everyone. I've been working with sails 0.10.2 version and sometimes when I quit my browser and restart it to hit my app url , I get an error like this:

error: Could not load session for socket #F4O-lvWICcs6uQTYREOc
error: The socket's cookie could not be parsed into a sessionID.
error: Unless you're overriding the `authorization` function, make sure you pass in a valid `sails.sid` cookie
error: (or omit the cookie altogether to have a new session created and an encrypted cookie sent in the response header to your socket.io upgrade request)
error: TypeError: Cannot call method 'indexOf' of undefined
    at exports.parseSignedCookie (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/node_modules/express/node_modules/connect/lib/utils.js:199:19)
    at Object.Session.fromSocket (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/lib/hooks/session/index.js:339:38)
    at interpretSocketReq (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/lib/hooks/sockets/lib/interpreter/interpret.js:542:19)
    at Socket.<anonymous> (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/lib/hooks/sockets/lib/connection.js:95:5)
    at Socket.emit [as $emit] (events.js:98:17)
    at SocketNamespace.handlePacket (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/node_modules/socket.io/lib/namespace.js:335:22)
    at Manager.onClientMessage (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/node_modules/socket.io/lib/manager.js:518:38)
    at WebSocket.Transport.onMessage (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/node_modules/socket.io/lib/transport.js:369:20)
    at Parser.<anonymous> (/Users/bshen/Desktop/educloud-v20/backend/node_modules/sails/node_modules/socket.io/lib/transports/websocket/hybi-16.js:39:10)
    at Parser.emit (events.js:95:17) [TypeError: Cannot call method 'indexOf' of undefined]

I've tried to debug but can't figure out what's wrong. Any kind of request to my Sails API launches this error and I have to make a refresh/F5 on my browser to make it work again.

Does anyone have an idea?
Regards,

**Comment by stensrud** [comment]
+1

**Comment by GeoCloud** [comment]
+1

**Comment by thiagof** [comment]
Have you tried to enable session storage?

2015-02-17 19:13 GMT-02:00 GeoCloud <notifications@github.com>:

> +1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balderdashy/sails/issues/2263#issuecomment-74754617>.
>



-- 
*Thiago Fernandes*
Analista de Processos
34 9176- 4055 | thiago@internetbudi.com.br

**Comment by JetFault** [comment]
I'm getting the same thing. I overwrote my socket.js `authorization` to use my own logic. And according to the error message, that's one way to remove this issue, but that doesn't seem to be the case.

I'm wondering is there a way to turn off sessions in general (I'm using JWT). I've disabled the cookieParser middleware, but now want to also disable all sessions in generale.

**Comment by mikermcneil** [comment]
Sorry for the slow response guys.

@JetFault re: disabling session- you can do so by disabling the session hook.  As of Sails 0.11, the sockets hook should fully support session-free usage.

@stenstrud  @RobiFerentz @svnh @billyshena have you been able to reproduce this in Sails v0.11 w/ the new sockets hook?

**Comment by mackenrou** [comment]
Hi @mikermcneil 
I updated to sails 0.11 disabling the session hook and fix this, but now socket don't connect on android browser and phonegap app as i explained here: https://github.com/balderdashy/sails/issues/2874

**Comment by thiagof** [comment]
Best solution to me were using a database session adapter

2015-04-27 11:09 GMT-03:00 mackenrou <notifications@github.com>:

> Hi @mikermcneil <https://github.com/mikermcneil>
> I updated to sails 0.11 disabling the session hook and fix this, but now
> socket don't connect on android browser and phonegap app as i explained
> here: #2874 <https://github.com/balderdashy/sails/issues/2874>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balderdashy/sails/issues/2263#issuecomment-96667459>.
>



-- 
*Thiago Fernandes*
Analista de Processos
34 9176- 4055 | thiago@internetbudi.com.br

**Comment by mikermcneil** [comment]
Thanks for posting, @billyshena.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
